apps/dashboard/src/components/billing/hooks/use-subscription.ts

```diff
@@ -0,0 +1,84 @@
+import { useQuery } from '@tanstack/react-query';
+import { ApiServiceLevelEnum } from '@novu/shared';
+import { get } from '../../../api/api.client';
+import { differenceInDays, isSameDay } from 'date-fns';
+import { useMemo } from 'react';
+
+export interface UseSubscriptionType {
+  isLoading: boolean;
+  data: {
+    apiServiceLevel: ApiServiceLevelEnum;
+    isActive: boolean;
+    hasPaymentMethod: boolean;
+    status: string;
+    currentPeriodStart: string | null;
+    currentPeriodEnd: string | null;
+    billingInterval: 'month' | 'year' | null;
+    events: {
+      current: number;
+      included: number;
+    };
+    trial: {
+      isActive: boolean;
+      start: string;
+      end: string;
+      daysTotal: number;
+      daysLeft: number;
+    };
+  };
+}
+
+type SubscriptionResponse = Omit<UseSubscriptionType, 'isLoading'>;
+
+export function useSubscription(): UseSubscriptionType {
+  const { data, isLoading } = useQuery<SubscriptionResponse>({
+    queryKey: ['subscription'],
+    queryFn: () => get('/billing/subscription'),
+  });
+
+  const daysLeft = useMemo(() => {
+    const today = new Date();
+    if (!data?.data.trial.end) return 0;
+
+    return isSameDay(new Date(data.data.trial.end), today) ? 0 : differenceInDays(new Date(data.data.trial.end), today);
+  }, [data?.data.trial.end]);
+
+  if (isLoading || !data) {
+    return {
```

Comment:
It's a good opportunity to refactor this hook a little bit and make it work with a subscription object from the database or null. Null in data means there is no subscription yet, and the page needs to load. 

This default state creates a duplication of code between the frontend and the API that we can remove.

Reply:
I agree, first wanted to make it work 😄 Going to refactor a few of those older pieces 💪